src/client/common/application/encryptedStorage.ts

```diff
@@ -40,8 +45,12 @@ export class EncryptedStorage implements IEncryptedStorage {
             return val;
         } catch (e) {
             // If we get an error trying to get a secret, it might be corrupted. So we delete it.
-            await this.extensionContext.secrets.delete(`${service}.${key}`);
-            return;
+            try {
+                this.extensionContext.secrets.delete(`${service}.${key}`);
```

Comment:
I believe this won't be caught here. The promise is ignored.

You might try this instead:
```
this.extensionContext.secrets.delete(`${service}.${key}`).catch(traceError);
```

Reply:
thenables don't have .catch

Reply:
I'll put the await back